src/main_thread/init/multi_thread_content_initializer.ts

```diff
@@ -1121,10 +1122,9 @@ export default class MultiThreadContentInitializer extends ContentInitializer {
           if (this._currentContentInfo?.contentId !== msgData.contentId) {
             return;
           }
-          const resolveFn = this._segmentMetrics.resolvers[msgData.value.messageId];
+          const resolveFn = this._segmentMetrics.resolvers.get(msgData.value.messageId);
           if (resolveFn !== undefined) {
             resolveFn(msgData.value.segmentSinkMetrics);
-            delete this._segmentMetrics.resolvers[msgData.value.messageId];
           } else {
```

Comment:
the `delete` is removed, I think it  need to be freed when the message has been handled ?

Reply:
delete is done closer to where definition is done in the `resolveFn` function. I thought that it made more sense